smithy4s/src/main/scala-2/com/dwolla/consul/smithy4s/ConsulDiscoverablePlatform.scala

```diff
@@ -0,0 +1,77 @@
+package com.dwolla.consul.smithy4s
+
+import cats.syntax.all._
+import com.dwolla.consul.smithy._
+import org.http4s.Uri.{Host, Scheme}
+import org.typelevel.scalaccompat.annotation.nowarn212
+import smithy4s.Hints
+
+import scala.reflect.macros.whitebox
+import scala.util.Try
+
+trait ConsulDiscoverablePlatform {
+  implicit def derivedInstance[Alg[_[_]]]: ConsulDiscoverable[Alg] =
+    macro ConsulDiscoverableMacros.makeInstance[Alg]
+}
+
+object ConsulDiscoverableMacros {
+  @nowarn212("msg=local val (liftableScheme|liftableHost) in method makeInstance is never used")
+  def makeInstance[Alg[_[_]]](c: whitebox.Context): c.Expr[ConsulDiscoverable[Alg]] = {
+    import c.universe.{Try => _, _}
+
+    implicit val liftableScheme: Liftable[Scheme] = Liftable { scheme: Scheme =>
+      q"""_root_.org.http4s.Uri.Scheme.unsafeFromString(${scheme.value})"""
+    }
+    implicit val liftableHost: Liftable[Host] = Liftable { host: Host =>
+      q"""_root_.org.http4s.Uri.Host.unsafeFromString(${host.value})"""
+    }
+
+    val consulScheme = DiscoveryMacros.consulScheme.some
+
+    def findHintsInTree(tpe: Tree): Either[String, Hints] =
+      Try {
+        tpe.collect {
+            case x: TypTree =>
+              c.eval(c.Expr[Hints](q"${x.symbol.companion}.hints"))
+          }
+          .headOption
+          .toRight(s"could not find hints for $tpe")
+      }
+        .toEither
+        .leftMap(_.toString ++ s" (is $tpe a Smithy4s Service?)")
+        .flatten
+
+    def getDiscoverableFromHints(tpe: Tree): Hints => Either[String, Discoverable] =
+      _.get(Discoverable.tagInstance)
+        .toRight(s"could not find Discoverable hint for $tpe")
+
+    val getHostFromDiscoverable: PartialFunction[Discoverable, Either[String, Host]] = {
+      case Discoverable(ServiceName(serviceName)) =>
+        Host.fromString(serviceName)
+          .leftMap(_.message)
+    }
+
+    def hostToConsulDiscoverableExpr(tpe: Tree): Host => c.Expr[ConsulDiscoverable[Alg]] = host =>
+      c.Expr[ConsulDiscoverable[Alg]](
+        q"""
+          new _root_.com.dwolla.consul.smithy4s.ConsulDiscoverable[$tpe] {
```

Comment:
My mental model of how macros work is that it's like the code block is copy/pasted into the spot where the macro is invoked. The imports, etc., that are in effect in that spot are what controls how types are resolved. So e.g. a prior version used the `.some` enhancement method (`uriAuthority.some`) instead of wrapping the value in an `Option` as below, which worked in places where the cats syntax had been imported, but failed otherwise. 

To avoid that kind of issue or other issues with ambiguous imports, I fully qualified everything in the quasiquote so that it should work regardless of context.